### PR TITLE
Fix jet matching with Pythia Versions later than 8.3

### PR DESCRIPTION
**Pull request by bistapf**
BEGINRELEASENOTES
- Add the missing `else` block to the  `DelphesPythia8Reader` to make jet matching also work with Pythia Versions later than 8.3. Use same code as in the equivalent [Delphes reader](https://github.com/delphes/delphes/blob/master/readers/DelphesPythia8.cpp#L301). 

ENDRELEASENOTES

### CODE REVIEW

File: standalone/src/DelphesPythia8Reader.h [hunk 1]
```diff
@@ -58,11 +58,17 @@ class DelphesPythia8Reader : public DelphesInputReader {
 
     // jet matching
 #if PYTHIA_VERSION_INTEGER < 8300
+    Pythia8::CombineMatchingInput* combined   = 0;
+    Pythia8::UserHooks*            m_matching = 0;
```
Comment (tmadlener) [inline standalone/src/DelphesPythia8Reader.h:62]:
```suggestion
    Pythia8::UserHooks*            matching = 0;
```
Can this variable be renamed to not have the `m_` prefix as that implies it to be a member variable, but it is just a local variable in this function.

**Comment by tmadlener** [comment]
Is there an easy way to check whether matching works as expected that could be put into a test case?

**Comment by bistapf** [comment]
Do you mean to check if it's actually running/properly set-up, or also that it gives correct physics?

For the first, can just check if: `number reco events < number input LHE events` in a test case with matching turned on. Should always be the case if jet matching works correctly I think(note that only for some FCC-hh sample we need to use the matching, so need a dedicated tester job with one of these). And/or could check for this print-out at the end of the run:
```
 *-------  JetMatching Error and Warning Messages Statistics  -----------------------------------------------------* 
 |                                                                                                                 | 
 |  times   message                                                                                                | 
 |                                                                                                                 | 
 |      0   no errors or warnings to report                                                                        | 
 |                                                                                                                 | 
 *-------  End JetMatching Error and Warning Messages Statistics  -------------------------------------------------* 
```
Don't know if `Pythia` has some internal variable it uses for this that we could access/check. 

To check also if physics makes sense, could compare the resulting matching efficiency of a test job to whatever this process had in a previous production (this is stored in the FCC database somewhere), but they are derived from very large scale productions so would be a rough comparison since it might vary a little depending which LHE file/seed you're running on.

**Comment by tmadlener** [comment]
No, really just on the technical level to make sure if a pythia card with matching turned on, we actually get matching running. So checking the output is a solution, but we should make sure that the output was indeed different before this fix (and that the test actually fails before).

**Review by tmadlener — APPROVED** [review]

**Pull request merged**